Sources/Parsing/Parsers/Whitespace.swift

```diff
@@ -1,64 +1,132 @@
-/// A parser that consumes all ASCII whitespace from the beginning of the input.
-///
-/// - Note: This parser only consumes ASCII spaces (`" "`), newlines (`"\n"` and `"\r"`), and tabs
-///   (`"\t"`). If you need richer support that covers all unicode whitespace, use a ``Prefix``
-///   parser that operates on the `Substring` level with a predicate that consumes whitespace:
-///
-///   ```swift
-///   Prefix { $0.isWhitespace }
-///   ```
+/// A parser that consumes whitespace from the beginning of input.
 public struct Whitespace<Input: Collection, Bytes: Collection>: Parser
 where
   Input.SubSequence == Input,
-  Bytes.SubSequence == Bytes,
-  Bytes.Element == UTF8.CodeUnit
+  Bytes.Element == UTF8.CodeUnit,
+  Bytes.SubSequence == Bytes
 {
+  public enum Configuration {
+    case all
+    case horizontal
+    case vertical
+  }
+
+  public let configuration: Configuration
+
   @usableFromInline
   let toBytes: (Input) -> Bytes
 
   @usableFromInline
   let fromBytes: (Bytes) -> Input
 
   @inlinable
-  public func parse(_ input: inout Input) -> Input {
-    let output = self.toBytes(input).prefix(while: { (byte: UTF8.CodeUnit) in
-      byte == .init(ascii: " ")
-        || byte == .init(ascii: "\n")
-        || byte == .init(ascii: "\r")
-        || byte == .init(ascii: "\t")
-    })
-    input.removeFirst(output.count)
-    return self.fromBytes(output)
-  }
-}
+  public func parse(_ input: inout Input) {
```

Comment:
There may be better and more performant ways of defining the below, but it seems straightforward and doesn't seem to have a huge impact on existing performance (it is slightly slower, though)